Source/WebKit/WebProcess/WebPage/WebPage.messages.in

```diff
@@ -47,8 +47,8 @@ messages -> WebPage LegacyReceiver {
     ViewWillEndLiveResize()
 
     ExecuteEditCommandWithCallback(String name, String argument) -> ()
-    KeyEvent(WebKit::WebKeyboardEvent event)
-    MouseEvent(WebKit::WebMouseEvent event, std::optional<Vector<WebKit::SandboxExtension::Handle>> sandboxExtensions)
+    KeyEvent(std::optional<WebCore::FrameIdentifier> frameID, WebKit::WebKeyboardEvent event)
+    MouseEvent(std::optional<WebCore::FrameIdentifier> frameID, WebKit::WebMouseEvent event, std::optional<Vector<WebKit::SandboxExtension::Handle>> sandboxExtensions)
```

Comment:
I had this initially, but I changed it because there was cases where the main frame didn’t exist, but we still expected didReceiveEvent to be called from the web process to indicate that the event was not handled.

TestWebKitAPI.WebKit.DidNotHandleKeyDown was the relevant test.

I can look to see if there is a better way to handle that situation.

Reply:
I see, that's indeed a bit subtle!

Perhaps a simpler alternative would be to just skip the round trip and consider the event handled in the UI process if there's no frame that's capable of handling the event — that said, I think the `std::optional<>` should be okay too (maybe with a brief comment explaining when the ID is `std::nullopt`).